src/index.js

```diff
@@ -19,41 +18,98 @@ const showList = (array) => {
             class="checkbox" 
             name="checkbox" 
             id="checkbox" ></li>
-          <li class="task-description">${array.description}</li>
+            <input type="text" class="task-description" name="${array.description}" id="${array.index + 1}" value="${array.description}">
         </ul>
       </div>
 
     <div class="actions">
-      <i class="fa-solid fa-pen-to-square edit"></i>
+      <i class="fa-solid fa-pencil"></i>
+      <i class="fa-solid fa-floppy-disk edit"></i>
       <i class="fa-solid fa-trash-can del"></i>
     </div>
     </div>
       `;
   return container;
 };
 
-const addlist = (array) => {
+const addNew = (array, low) => {
+  // eslint-disable-next-line no-const-assign
   const taskN = {};
-  taskN.description = newDest.value;
-  taskN.completed = array;
-  taskN.index = task.length + 1;
-  const listItem = showList(taskN);
-  newDest.insertAdjacentHTML('beforeend', listItem);
-  task.push(taskN);
-  addTask.innerHTML += listItem;
+  taskN.description = array;
+  taskN.completed = false;
+  taskN.index = low.length + 1;
+  low.push(taskN);
+  localStorage.setItem('tasks', JSON.stringify(low));
+};
+const addList = () => {
+  let tasks = JSON.parse(localStorage.getItem('tasks'));
+  if (tasks == null) {
+    localStorage.setItem('tasks', JSON.stringify([]));
+  }
+  tasks = JSON.parse(localStorage.getItem('tasks'));
+  if (newDest.value === '') {
+    // eslint-disable-next-line no-alert
+    window.alert('can not save, empty value');
+    return;
+  }
+  addNew(newDest.value, tasks);
+};
+
+const deleteOne = (array) => {
+  let taskName = array.children[0].children[0].children[1].id;
+  taskName = parseInt(taskName, 10);
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  const ele = tasks.findIndex((e) => e.index === taskName);
+  tasks.splice(ele - 1, 1);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
```

Comment:
- [ ] After removing a task, kindly update the indexes of the remaining tasks so they represent the current list order and are unique. Example
```javascript
tasks = [
  { 
     description: task1,
    index: 1
  },
  { 
     description: task2,
    index: 2
  },
  { 
     description: task3,
    index: 3
  }
]
```
If the task with index: 2 is removed. The indexes should be updated like;
```javascript
tasks = [
  { 
     description: task1,
    index: 1
  },
  { 
     description: task3,
    index: 2
  }
]
```